tools/bake.py

```diff
@@ -572,9 +756,14 @@ def first_bsdf(objs):
                             idx = child.data.uv_layers.active_index
                             child.data.uv_layers[layer].active = True
                             Common.switch('EDIT')
-                            bpy.ops.mesh.select_all(action='SELECT')
-                            bpy.ops.uv.select_all(action='SELECT')
-                            bpy.ops.uv.smart_project(angle_limit=66.0, island_margin=0.01)
+                            for matindex,material in enumerate(child.data.materials):
+                                #select each material individually and unwrap. The averaging and packing will take care of overlaps. - @989onan
+                                bpy.ops.mesh.select_all(action='DESELECT')
+                                child.active_material_index = matindex
+                                bpy.ops.object.material_slot_select()
+                                
+                                bpy.ops.uv.select_all(action='SELECT')
+                                bpy.ops.uv.smart_project(angle_limit=89.0, island_margin=margin)
```

Comment:
I had pretty bad distortion issues when using a smart project angle > 60, not sure if this is a related change?

Also it's probably better to make this block an option...

Reply:
I can make this block an option in the next commit real quick